packages/harpguru-core/src/hooks/use-flush-buffered-activity-toggles/use-flush-buffered-activity-toggles.ts

```diff
@@ -0,0 +1,133 @@
+  // fails to return a different value to the *current* value on
+  // each render. Possibly what's happening is something like
+  // the `usePrevious` is *seeing* each update to the variables,
+  // but is only getting to render on the second one because they
+  // happen in such quick succession. Very speculative and I'd be
+  // interested to discover the truth.
+
+  // It is also a complete mystery why this problem was introduced at
+  // commit 9a2d523b43741ae3bf44b390b3371cb3a7d38a08 when only the
+  // quiz cycle component was modified there and the problem isn't
+  // resolved if you remove that component from the app entirely..
+  // It's an even bigger surprise when you consider that the previous
+  // commit at 49c0e580d532a675586be6e4880b5ab31251594b removed the
+  // batch processing from this toggle flush hook, which seems like
+  // something which would be much more likely to be the cause of
+  // such an issue.
+  unstable_batchedUpdates(() => {
+    emptyBufferedToggles()
+    updateActiveHarpStrata(bufferedActivityToggles)
+  })
+}
+
+export const useFlushBufferedActivityToggles = (): ((
+  arg0: ReadonlyArray<DegreeIds>
+) => void) => {
+  const updateActiveHarpStrata = useDispatch(
+    (
+      activeHarpStrata: HarpStrata,
+      bufferedActivityToggles: ReadonlyArray<DegreeIds>
+    ): HarpStrata => {
+      const {
+        apparatus: { id: apparatusId },
+        pozitionId,
+        harpKeyId,
+        activeDegreeIds,
+      } = activeHarpStrata
+      const newActiveDegreeIds = batchToggleDegreeIds(
+        activeDegreeIds,
+        bufferedActivityToggles
+      )
+      const newHarpStrataProps: HarpStrataProps = {
+        apparatusId,
+        pozitionId,
+        harpKeyId,
+        activeIds: newActiveDegreeIds,
+      }
+      return getHarpStrata(newHarpStrataProps)
+    },
+    'activeHarpStrata'
+  )
+
+  const emptyBufferedToggles = useDispatch(
+    (): [] => [],
+    'bufferedActivityToggles'
+  )
+
+  return (bufferedActivityToggles) =>
+    flushBufferedActivityToggles(
+      updateActiveHarpStrata,
+      emptyBufferedToggles,
+      bufferedActivityToggles
+    )
+}
+
+export const useFlushBufferedActivityTogglesSingleDispatch = (): (() => void) => {
+  const updateActiveHarpStrata = useDispatch((global: GlobalState) => {
+    const { activeHarpStrata, bufferedActivityToggles } = global
+
+    if (bufferedActivityToggles.length === 0) return
+
+    const {
+      apparatus: { id: apparatusId },
+      pozitionId,
+      harpKeyId,
+      activeDegreeIds,
+    } = activeHarpStrata
+    const newActiveDegreeIds = batchToggleDegreeIds(
+      activeDegreeIds,
+      bufferedActivityToggles
+    )
+    const newHarpStrataProps: HarpStrataProps = {
+      apparatusId,
+      pozitionId,
+      harpKeyId,
+      activeIds: newActiveDegreeIds,
+    }
+
+    return {
+      activeHarpStrata: getHarpStrata(newHarpStrataProps),
+      bufferedActivityToggles: [],
+    }
+  })
+
+  return () => {
```

Comment:
This needs a comment explaining why it needs to be wrapped in a batch update. The reason is recorded in the commit message where this was included. I could raise an issue to understand why this is.

I could also consider replacing this wrapper with a pattern whereby the wrapper must be used if the function is being called from a `setTimeout` context.

Reply:
Comment added and https://github.com/js-jslog/harpguru/issues/62 raised